code/datums/ai/dog/dog_behaviors.dm

```diff
@@ -168,7 +168,7 @@
 /datum/ai_behavior/harass/perform(delta_time, datum/ai_controller/controller)
 	. = ..()
 	var/mob/living/living_pawn = controller.pawn
-	if(!istype(living_pawn))
+	if(!istype(living_pawn) || !(isturf(living_pawn.loc) || HAS_TRAIT(living_pawn, TRAIT_AI_BAGATTACK)))
```

Comment:
you could probably add some logic similar to (but not quite the same as) that of stabilized extracts to let dogs that are held in someone's hand (or pocket) attack people, but not dogs that are inside of an object or something. something like a check to see if the dog is inside of a mob holder, and then an isliving(living_pawn.loc.loc) check or something